src/algebra/associated.lean

```diff
@@ -316,6 +316,15 @@ lemma prime_of_associated [comm_monoid_with_zero α] {p q : α} (h : p ~ᵤ q) (
     ⟨λ ⟨v, hv⟩, hp.not_unit ⟨v * u⁻¹, by simp [hv, hu.symm]⟩,
       hu ▸ by { simp [units.mul_right_dvd], intros a b, exact hp.div_or_div }⟩⟩
 
+lemma primes_associated_of_dvd {α} [comm_cancel_monoid_with_zero α] {p q: α}
+  (p_prime: prime p) (q_prime: prime q) (dvd: p ∣ q): associated p q :=
+begin
+  cases dvd with c hc,
+  cases ((irreducible_of_prime q_prime).is_unit_or_is_unit hc)
+    .resolve_left p_prime.not_unit with u hu,
+  exact ⟨ u, by rw [hu, hc] ⟩,
+end
```

Comment:
I'm going to be annoying and make a suggestion that basically replaces all your code, because I saw a nicer way to do it.

My thought process: I recalled that `associated p q` means `p | q ∧ q | p`, so it should suffice to show `p | q → q | p`. It turns out we already had that result, called `dvd_symm_of_irreducible`. So we can even generalize to `_of_irreducible` and get `_of_prime` as a special case.

```suggestion
lemma associated_of_irreducible_of_dvd [comm_cancel_monoid_with_zero α] {p q : α}
  (p_irr : irreducible p) (q_irr : irreducible q) (dvd : p ∣ q) : associated p q :=
associated_of_dvd_dvd dvd (dvd_symm_of_irreducible p_irr q_irr dvd)

lemma associated_of_prime_of_dvd [comm_cancel_monoid_with_zero α] {p q : α}
  (p_prime : prime p) (q_prime : prime q) (dvd : p ∣ q) : associated p q :=
associated_of_irreducible_of_dvd (irreducible_of_prime p_prime) (irreducible_of_prime q_prime) dvd
```

Reply:
Ok, this is better, thanks !